src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -848,92 +667,90 @@ public void SetMetadataEmitter(MetadataWriter metadataWriter)
 
         public unsafe BlobContentId GetContentId()
         {
+            BlobContentId contentId;
+
             if (_deterministic)
             {
-                // rewrite GUID and timestamp in the PDB with hash of a has of the log content:
-                byte[] hash = _callLogger.GetLogHash();
+                // Flush any remaining data to the underlying stream.
+                _symWriter.Commit();
 
-                try
-                {
-                    fixed (byte* hashPtr = &hash[0])
-                    {
-                        ((ISymUnmanagedWriter8)_symWriter).UpdateSignatureByHashingContent(hashPtr, hash.Length);
-                    }
-                }
-                catch (Exception ex)
-                {
-                    throw new PdbWritingException(ex);
-                }
-            }
+                // Calculate hash of the stream content.
+                // Note: all bits of the signature currently stored in the PDB stream were initialized to 1 by InitializeDeterministic.
+                contentId = BlobContentId.FromHash(CryptographicHashProvider.ComputeSha1(_pdbStream.GetChunks()));
 
-            // See symwrite.cpp - the data byte[] doesn't depend on the content of metadata tables or IL.
-            // The writer only sets two values of the ImageDebugDirectory struct.
-            // 
-            //   IMAGE_DEBUG_DIRECTORY *pIDD
-            // 
-            //   if ( pIDD == NULL ) return E_INVALIDARG;
-            //   memset( pIDD, 0, sizeof( *pIDD ) );
-            //   pIDD->Type = IMAGE_DEBUG_TYPE_CODEVIEW;
-            //   pIDD->SizeOfData = cTheData;
-
-            ImageDebugDirectory debugDir = new ImageDebugDirectory();
-            uint dataLength;
-
-            try
-            {
-                _symWriter.GetDebugInfo(ref debugDir, 0, out dataLength, IntPtr.Zero);
+                ((ISymUnmanagedWriter8)_symWriter).UpdateSignature(contentId.Guid, contentId.Stamp, age: 1);
```

Comment:
The try/catch was removed. This can't throw any longer?

Reply:
Good catch. It can throw. 

---
In reply to: [137341233](https://github.com/dotnet/roslyn/pull/21712#discussion_r137341233) [](ancestors = 137341233)